core/src/main/java/hudson/security/HttpSessionContextIntegrationFilter2.java

```diff
@@ -84,7 +84,13 @@ private boolean hasInvalidSessionSeed(Authentication authentication, HttpSession
             return false;
         }
 
-        User userFromSession = User.getById(authentication.getName(), false);
+        User userFromSession;
+        try {
+            userFromSession = User.getById(authentication.getName(), false);
+        } catch (IllegalStateException ise) {
+            logger.warn("Encountered IllegalStateException trying to get a user. System init may not have completed yet. Invalidating user session.");
```

Comment:
Would it be useful to include the exception in the log message?

Reply:
I debated that myself. My opinion is that because the handling is so narrow, it would just introduce clutter to include the exception. If we broadened the catch, then we would definitely want to add the exception.

Reply:
It the issue is understood, a stack trace only makes the log look scarier. It's easy to misread a stack trace as an exception that wasn't handled properly.

Reply:
@jvz do you consider this comment as resolved?

Reply:
Oh yeah, good with me.